File: AgentUp.Desktop/Features/FakeServer/Controllers/FakeServerController.cs
using AgentUp.Desktop.Features.FakeServer.DTOs;
using AgentUp.Desktop.Features.FakeServer.Models;
using AgentUp.Desktop.Features.FakeServer.Services;

namespace AgentUp.Desktop.Features.FakeServer.Controllers;

public sealed class FakeServerController(FakeBackendService backend)
{
    public FakeServerConnectionDto Catalog(string? activeServerId)
        => backend.Catalog(activeServerId);

    public bool Matches(Uri? uri)
        => backend.Matches(uri);

    public bool Matches(string? url)
        => backend.Matches(url);

    public string? ApplicationHtml(int allocatedPort)
        => backend.ApplicationHtml(allocatedPort);

    public void Reset()
        => backend.Reset();
}

File: AgentUp.Mobile/src/features/fake-server/controllers/FakeServerController.ts
import { fakeServerEntry } from '../providers/FakeServerCatalogProvider';
import { loadFakeServerDefinition } from '../providers/FakeServerDefinitionProvider';
import { installFakeServerFetch, uninstallFakeServerFetch } from '../providers/FakeServerFetchProvider';
import { FakeBackendService } from '../services/FakeBackendService';

const backend = new FakeBackendService(loadFakeServerDefinition());

export class FakeServerController {
  catalog(activeServerId?: string | null) {
    return fakeServerEntry({
      servers: [],
      activeServerId: activeServerId ?? null,
    });
  }

  matches(url: string | null | undefined) {
    return backend.matches(url);
  }

  applicationHtml(allocatedPort: number) {
    return backend.applicationHtml(allocatedPort);
  }

  activate() {
    backend.reset();
    installFakeServerFetch(backend);
  }

  deactivate() {
    uninstallFakeServerFetch();
  }

  reset() {
    backend.reset();
  }
}

export const fakeServers = new FakeServerController();
